src/app/profile/edit-profile/edit-profile.component.ts

```diff
@@ -1,10 +1,68 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { Profile } from '../../shared/models/profile.model';
+import { ProfileService } from './profile.service';
+import { Router } from '@angular/router';
+import { environment } from '../../../env/environment';
 
 @Component({
   selector: 'app-edit-profile',
   templateUrl: './edit-profile.component.html',
-  styleUrl: './edit-profile.component.css'
+  styleUrls: ['./edit-profile.component.css']
 })
-export class EditProfileComponent {
+export class EditProfileComponent implements OnInit {
+  profile: Profile | null;
 
+  constructor(
+    private profileService: ProfileService,
+    private router: Router
```

Comment:
Add shared service so we can open snackbars.
```suggestion
    private router: Router,
    private sharedService: SharedService
```